src/crypto/qat/qcccrypto.cc

```diff
@@ -121,12 +186,16 @@ bool QccCrypto::init()
     this->cleanup();
     return false;
   }
+  dout(1) << "Get instances num: " << qcc_inst->num_instances << dendl;
+  max_queue_size = max_requests - qcc_inst->num_instances;
+  instance_completions.set_capacity(max_queue_size);
```

Comment:
what if `max_requests` <= `num_instances`? should we leave `instance_completions` empty so callers never wait?
```c++
if (max_requests > qcc_inst->num_instances) {
  instance_completions.set_capacity(max_requests - qcc_inst->num_instances);
}
```
or would it make sense to set some minimum value so there's always *some* room for waiters?

Reply:
> what if `max_requests` <= `num_instances`? should we leave `instance_completions` empty so callers never wait?
> 
> ```c++
> if (max_requests > qcc_inst->num_instances) {
>   instance_completions.set_capacity(max_requests - qcc_inst->num_instances);
> }
> ```
> 
> or would it make sense to set some minimum value so there's always _some_ room for waiters?

This is a good idea. When `max_requests` <= `num_instances`, we can leave `instance_completions` empty.